lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -115,7 +115,7 @@ class _Visitor extends SimpleAstVisitor<void> {
       // The parens cannot be removed because then the curly brackets are not
       // interpreted as a set-or-map literal.
       if (parent is PropertyAccess || parent is MethodInvocation) {
-        Expression target = (parent as dynamic).target;
+        var target = (parent as dynamic).target;
```

Comment:
Normally I would not like this fix, as it might invoke dynamic later (just a personal distaste), but I see its just doing an `==`, so its clean enough. My target access is kind of  hacky anyway. PropertyAccess and MethodInvocation both just _happen_ to have a field named target, and I took advantage of that.

Reply:
An alternative would be to introduce a `_getTarget` helper that returned the target or `null`.

Reply:
Yeah.  It's not an ideal fix.

> Do you have no-implicit-casts turned on or something, in your IntelliJ?

Actually we do in analysis options.

https://github.com/dart-lang/linter/blob/f3413e972bbe8015009ec1450245cc63d51963bf/analysis_options.yaml#L5

It's a puzzle as to why travis isn't flagging it without the explicit add.